src/Core/Http/SecureHttpClient.php

```diff
@@ -136,17 +138,77 @@ public function signedRequest($method, $endpoint, array $payload = [], $returnJs
         }
     }
 
+    private function getAlg()
+    {
+        $privateKey = $this->accountKeyPair->getPrivateKey();
+        $parsedKey = $this->keyParser->parse($privateKey);
+        switch ($parsedKey->getType()) {
+            case OPENSSL_KEYTYPE_RSA:
+                return 'RS256';
+            case OPENSSL_KEYTYPE_EC:
+                switch ($parsedKey->getBits()) {
+                    case 256:
+                    case 384:
+                        return 'ES'.$parsedKey->getBits();
+                    case 521:
+                        return 'ES512';
+                }
+                throw new AcmeCoreClientException('Private key type is not supported');
```

Comment:
I think if you don't put any `break`, default will be executed, so this line is not needed IMO.

Reply:
done, even when I don't really like case without break